browser/src/control/Control.PartsPreview.js

```diff
@@ -617,29 +617,10 @@ L.Control.PartsPreview = L.Control.extend({
 
 	_onScroll: function (e) {
 		setTimeout(L.bind(function (e) {
-			var scrollOffset = 0;
-			if (e) {
-				var prevScrollY = this._scrollY;
-				var rectangle = e.target.getBoundingClientRect();
-				this._scrollY = this._direction === 'x' ? -rectangle.left : -rectangle.top;
-				scrollOffset = this._scrollY - prevScrollY;
-			}
-
-			var previewContBB = this._partsPreviewCont.getBoundingClientRect();
-			var extra =  this._direction === 'x' ? previewContBB.width : previewContBB.height;
-			var topBound = this._previewContTop - (scrollOffset < 0 ? extra : extra / 2);
-			var bottomBound = this._previewContTop + extra + (scrollOffset > 0 ? extra : extra / 2);
 			for (var i = 0; i < this._previewTiles.length; ++i) {
-				var img = this._previewTiles[i];
-				if (img && img.parentNode && !img.fetched) {
-					var previewFrameBB = img.parentNode.getBoundingClientRect();
-					if (this._direction === 'x') {
-						if ((previewFrameBB.left >= topBound && previewFrameBB.left <= bottomBound)
-						|| (previewFrameBB.right >= topBound && previewFrameBB.right <= bottomBound)) {
-							this._map.getPreview(i, i, this.options.maxWidth, this.options.maxHeight, {autoUpdate: this.options.autoUpdate});
-						}
-					} else if ((previewFrameBB.top >= topBound && previewFrameBB.top <= bottomBound)
-						|| (previewFrameBB.bottom >= topBound && previewFrameBB.bottom <= bottomBound)) {
+				if (this._isPreviewVisible(i)) {
```

Comment:
I presume the idea is to match the logic of what we put into the queue of previews to fetch with the logic of what we use to take previews out of the queue and request them from the server?

Reply:
well, after reading the logic, it is the same. yes